python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -451,3 +458,94 @@ def close(self) -> None:
         if self.is_local_node(self._reader_node_id):
             self.ensure_registered_as_reader()
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
+
+
+@PublicAPI(stability="alpha")
+class MultiChannel(ChannelInterface):
+    """
+    Can be used to send data to different readers via different channels.
+    For example, if the reader is in the same worker process as the writer,
+    the data can be sent via IntraProcessChannel. If the reader is in a different
+    worker process, the data can be sent via shared memory channel.
+
+    Args:
+        writer: The actor that may write to the channel. None signifies the driver.
+        readers: The actors that may read from the channel. None signifies
+            the driver.
+    """
+
+    def __init__(
+        self,
+        writer: Optional[ray.actor.ActorHandle],
+        readers: List[Optional[ray.actor.ActorHandle]],
+        _channel_dict: Optional[Dict[ray.ActorID, ChannelInterface]] = None,
+        _channels: Optional[List[ChannelInterface]] = None,
+    ):
+        self._writer = writer
+        self._readers = readers
+        # A dictionary that maps the actor ID to the channel object.
+        self._channel_dict = _channel_dict or {}
+        # A list of channel objects.
+        self._channels = _channels or []
+        # TODO (kevin85421): Currently, the out-of-band actor handle is not well
+        # supported for reference counting. Here, we store the actor handle in
+        # `self._self_actor` to ensure its lifetime is the same as the channel object
+        # as a workaround. We should fix this issue in the future.
+        self._self_actor = _get_self_actor()
+        if self._channels:
+            return
```

Comment:
Add a comment to explain why we return early here.

Reply:
Updated https://github.com/ray-project/ray/pull/45591/commits/980cc82e9624f151f827af90561601ef3a84c35c